javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -79,1587 +91,989 @@ class Formatter(
+    if (flags.upperCase) localeInfo.toUpperCase(str)
+    else str
 
-    private def transform_j(): Unit = {
-      val day = calendar.get(Calendar.DAY_OF_YEAR)
-      result.append(paddingZeros(day, 3))
-    }
+  /** This method ignores `flags.zeroPad` and `flags.upperCase`. */
+  private def padAndSendToDestNoZeroPad(flags: Flags,
+                                        width: Int,
+                                        str: String): Unit = {
 
-    private def transform_y(): Unit = {
-      val year = calendar.get(Calendar.YEAR) % 100
-      result.append(paddingZeros(year, 2))
-    }
+    val len = str.length
 
-    private def transform_Y(): Unit = {
-      val year = calendar.get(Calendar.YEAR)
-      result.append(paddingZeros(year, 4))
-    }
+    if (len >= width)
+      sendToDest(str)
+    else if (flags.leftAlign)
+      sendToDest(str, strRepeat(" ", width - len))
+    else
+      sendToDest(strRepeat(" ", width - len), str)
+  }
 
-    private def transform_C(): Unit = {
-      val year = calendar.get(Calendar.YEAR) / 100
-      result.append(paddingZeros(year, 2))
-    }
+  /** This method ignores `flags.upperCase`. */
+  private def padAndSendToDest(localeInfo: LocaleInfo,
+                               flags: Flags,
+                               width: Int,
+                               prefix: String,
+                               str: String): Unit = {
+
+    val len = prefix.length + str.length
+    if (len >= width)
+      sendToDest(prefix, str)
+    else if (flags.zeroPad)
+      sendToDest(prefix,
+                 strRepeat(localeInfo.zeroDigitString, width - len),
+                 str)
+    else if (flags.leftAlign)
+      sendToDest(prefix, str, strRepeat(" ", width - len))
+    else
+      sendToDest(strRepeat(" ", width - len), prefix, str)
+  }
 
-    private def transform_a(): Unit = {
-      val day = calendar.get(Calendar.DAY_OF_WEEK)
-      result.append(getDateFormatSymbols().getShortWeekdays()(day))
+  private def strRepeat(s: String, times: Int): String = {
+    var result: String = ""
+    var i              = 0
+    while (i != times) {
+      result += s
+      i += 1
     }
+    result
+  }
 
-    private def transform_A(): Unit = {
-      val day = calendar.get(Calendar.DAY_OF_WEEK)
-      result.append(getDateFormatSymbols().getWeekdays()(day))
-    }
+  def ioException(): IOException = lastIOException
 
-    private def transform_b(): Unit = {
-      val month = calendar.get(Calendar.MONTH)
-      result.append(getDateFormatSymbols().getShortMonths()(month))
-    }
+  def locale(): Locale = {
+    checkNotClosed()
+    formatterLocaleInfo.locale
+  }
 
-    private def transform_B(): Unit = {
-      val month = calendar.get(Calendar.MONTH)
-      result.append(getDateFormatSymbols().getMonths()(month))
+  def out(): Appendable = {
+    checkNotClosed()
+    if (dest eq null) {
+      dest = new java.lang.StringBuilder(stringOutput)
+      stringOutput == ""
```

Comment:
```suggestion
      stringOutput = ""
```
(looks like Scala.js has the same problem)